thentos-adhocracy/src/Thentos/Adhocracy3/Types.hs

```diff
@@ -127,28 +147,30 @@ instance ThentosErrorToServantErr DB where
         -- For errors specifically relevant to the A3 frontend we mirror the A3 backend errors
         -- exactly so that the frontend recognizes them
         f (ThentosA3ErrorCore e) = case e of
-            BadCredentials -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            BadCredentials -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "password"
                 "body"
                 "User doesn't exist or password is wrong")
-            UserEmailAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            UserEmailAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "data.adhocracy_core.sheets.principal.IUserExtended.email"
                 "body"
                 "The user login email is not unique")
-            UserNameAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            UserNameAlreadyExists -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "data.adhocracy_core.sheets.principal.IUserBasic.name"
                 "body"
                 "The user login name is not unique")
-            NoSuchPendingUserConfirmation -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            NoSuchPendingUserConfirmation -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "path"
                 "body"
                 "Unknown or expired activation path")
-            NoSuchThentosSession -> (Nothing, mkA3StyleServantErr err400 $ A3ErrorInfo
+            NoSuchThentosSession -> (Nothing, mkA3StyleServantErr err400 $ A3Error
                 "X-User-Token"
                 "header"
                 "Invalid user token")
             _ -> thentosErrorToServantErr (Just mkErr) e
 
+        f (GenericA3Error errMsg) =
+            (Nothing, err400 {errBody = encode errMsg, errHeaders = [contentTypeJsonHeader]} )
```

Comment:
When A3 is down (= not reachable or doesn't respond) we'll throw a 500 instead of this GenericA3Error anyway. To requests such as this one A3 *always* replies with 400 in case of problems, so 400 should be safe here.

In rare cases (= internal problem) A3 could respond with 500 instead of 400, but those cases will still be distinguishable by the details of the A3 error message which we pass along. Along, the A3 frontend doesn't care for such details anyway in the case of activation clicks. It just displays the same "sorry that went wrong" method in case of *any* errors without caring for such details as status code or error message.

So it won't make a difference here.